src/JSInterop/Microsoft.JSInterop/test/Infrastructure/ByteArrayJsonConverterTest.cs

```diff
@@ -0,0 +1,157 @@
+            var ex = Record.Exception(() => JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions));
+            Assert.IsAssignableFrom<JsonException>(ex);
+        }
+
+        [Fact]
+        public void Read_ReadsJson()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json = $"{{\"__byte[]\":0}}";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Equal(byteArray, deserialized);
+        }
+
+        [Fact]
+        public void Read_IfByteArraysIdAppearsMultipleTimesUseLastProperty()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json = $"{{\"__byte[]\":9120,\"__byte[]\":0}}";
+
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Act & Assert
+            Assert.Equal(byteArray, deserialized);
+        }
+
+        [Fact]
+        public void Read_ReturnsTheCorrectInstance()
+        {
+            // Arrange
+            // Track a few arrays and verify that the deserialized value returns the correct value.
+            var byteArray1 = new byte[] { 1, 5, 7 };
+            var byteArray2 = new byte[] { 2, 6, 8 };
+            var byteArray3 = new byte[] { 2, 6, 8 };
+
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray1);
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray2);
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray3);
+
+            var json = $"[{{\"__byte[]\":2}},{{\"__byte[]\":1}}]";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[][]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Same(byteArray3, deserialized[0]);
+            Assert.Same(byteArray2, deserialized[1]);
+        }
+
+        [Fact]
+        public void Read_ReadsJson_WithFormatting()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json =
+@$"{{
+            ""__byte[]"": 0
+        }}";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Equal(byteArray, deserialized);
+        }
+
+        [Fact]
+        public void WriteJsonTwice_KeepsObjectId()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+
+            // Act & Assert
+            for (var i = 0; i < 10; i++)
+            {
+                var json = JsonSerializer.Serialize(byteArray, JsonSerializerOptions);
+                Assert.Equal($"{{\"__byte[]\":{i + 1}}}", json);
+            }
+        }
```

Comment:
Not sure what we are testing here, it says WriteJsonTwice but its doing something 10 times. Also the id seems to be changing every time.

I'm also not sure about the behavior here, we are not trying to de-dupe arrays, are we?

Reply:
Thanks for pointing this out, the test name needed to be updated to `WriteJsonMultipleTimes_IncrementsByteArrayId`. 

That's correct, we don't de-dupe arrays.